### PR TITLE
Quickstart Guide

### PR DESCRIPTION
**Pull request by ZYJLiu**
Add a "quick start" guide page.

### CODE REVIEW

**Comment by jacobcreech** [comment]
What is the expectation on how long this quickstart guide will take someone to complete @ZYJLiu ?

**Comment by ZYJLiu** [comment]
Probably hour or less. Its mostly steps to copy/paste into Solana Playground, where explanations are optionally available under a toggle.

File: docs/intro/quick-start.md [hunk 1]
```diff
@@ -0,0 +1,2826 @@
...
+- to: The user's account (destination of funds)
+
+  ```rs filename="lib.ts"
+  let transfer_accounts = Transfer {
+      from: ctx.accounts.vault_account.to_account_info(),
+      to: ctx.accounts.user.to_account_info(),
+  };
+  ```
+
+The CpiContext specifies:
```
Comment (nickfrosty) [inline docs/intro/quick-start.md:2598]:
```suggestion
The `CpiContext` specifies:
```

File: docs/intro/quick-start.md [hunk 2]
```diff
@@ -0,0 +1,2826 @@
...
+  let cpi_context = CpiContext::new(
+      ctx.accounts.system_program.to_account_info(),
+      transfer_accounts,
+  ).with_signer(signer_seeds);
+  ```
+
+The transfer function then invokes the transfer instruction on the System
+Program, passing:
+
+- The cpi_context (program, accounts, and PDA signer)
```
Comment (nickfrosty) [inline docs/intro/quick-start.md:2614]:
```suggestion
- The `cpi_context` (program, accounts, and PDA signer)
```

File: docs/intro/quick-start.md [hunk 3]
```diff
@@ -0,0 +1,2826 @@
...
+
+```shell filename="Terminal" copy
+build
+```
+
+### 3. Redeploy Program
+
+After making these changes, we need to redeploy our updated program. This
+ensures that our modified program is available for testing. On Solana, updating
+a program simply requires deploying the compiled program at the same program ID.
```
Comment (nickfrosty) [inline docs/intro/quick-start.md:2638]: can you clarify that you also need to be the authority, which is normally your cli keypair

**Review by nickfrosty — CHANGES_REQUESTED** [review]
This is a great addition John! I love the use of the native `details` html elements to show the diff and exmplanation. Genius idea. I also love that you added the diffs!

I left some minor comments and nits as suggestions

On top of those, can you also:
- add some internal links to the core concepts pages of the docs when they are initially mentioned in this guide
- many of the rust code blocks set the file name to `lib.ts` not `lib.rs`. can you update those?
- many of the code blocks have a `copy` attribute passed in the metadata. This does not actually do anything since all code blocks will automatically have a copy to clipboard button rendered via solana-com

Semi-related: while reviewing I also notice some css formatting issue that I will open a PR to solana-com to fix that will make the details elements have correct spacing relative to other cotnent on the page

File: docs/intro/quick-start.md [hunk 4]
```diff
@@ -0,0 +1,2826 @@
...
+(devnet by default) at the bottom of the window.
+
+![Connected](/assets/docs/intro/quickstart/pg-connected.png)
+
+<Callout>
+  Your Playground Wallet will be saved in your browser's local storage. Clearing
+  your browser cache will remove your saved wallet.
+</Callout>
+
+### 2. Get Devnet Sol
```
[the remark this answers is not in the document]
Comment (ZYJLiu) [reply docs/intro/quick-start.md:70]: Seems like there's a parsing bug when using capital SOL, which also affects the url. Submitted the [issue here](https://github.com/solana-foundation/solana-com/issues/1616).

`/docs/intro/quick-start#2-get-devnet-so`

CleanShot 2024-07-15 at 16 24 06@2x

File: docs/intro/quick-start.md (hunk 3 above)
Comment (ZYJLiu) [reply docs/intro/quick-start.md:2657]: added

```
<details>
{<summary>Explanation</summary>}

Only the upgrade authority of the program can update it. The upgrade authority
is set when the program is deployed, and it's the only account with permission
to modify or close the program. If the upgrade authority is revoked, then the
program becomes immutable and can never be closed or upgraded.

When deploying programs on Solana Playground, your Playground wallet is the
upgrade authority for all your programs.

</details>

```

File: docs/intro/quick-start.md [hunk 5]
```diff
@@ -0,0 +1,2826 @@
...
+  Reminder to update the cluster (network) connection on the Explorer you are
+  using to match Solana Playground. The default cluster is devnet.
+</Callout>
+
+### 4. Close Program
+
+Lastly, the SOL allocated to the on-chain program can be fully recovered by
+closing the program.
+
+You can close a program by running the following command and specifying the
```
[the remark this answers is not in the document]
Comment (ZYJLiu) [reply docs/intro/quick-start.md:1133]: added
```
<details>
{<summary>Explanation</summary>}

Only the upgrade authority of the program can close it. The upgrade authority is
set when the program is deployed, and it's the only account with permission to
modify or close the program. If the upgrade authority is revoked, then the
program becomes immutable and can never be closed or upgraded.

When deploying programs on Solana Playground, your Playground wallet is the
upgrade authority for all your programs.

</details>
```

File: docs/intro/quick-start.md [hunk 6]
```diff
@@ -0,0 +1,2826 @@
...
+  toPubkey: receiver.publicKey,
+  lamports: 0.01 * LAMPORTS_PER_SOL,
+});
+
+const transaction = new Transaction().add(transferInstruction);
+
+const transactionSignature = await sendAndConfirmTransaction(
+  pg.connection,
+  transaction,
+  [sender],
```
[the remark this answers is not in the document]
Comment (ZYJLiu) [reply docs/intro/quick-start.md:514]: added link to fees core concepts page

```
For more details, refer to
the [Transactions and Instructions](/docs/core/transactions) and
[Fees on Solana](/docs/core/fees) pages.
```

**Review by nickfrosty — APPROVED** [review]

**Pull request merged**